src/core/utils/Histogram.hpp

```diff
@@ -230,24 +240,28 @@ template <typename T> void Histogram<T>::normalize() { do_normalize(); }
  *        Divide by total number of counts times the bin volume.
  */
 template <typename T> void Histogram<T>::do_normalize() {
-  T tot_count =
-      std::accumulate(m_hist.begin(), m_hist.end(), static_cast<T>(0.0));
   T bin_volume = std::accumulate(m_bin_sizes.begin(), m_bin_sizes.end(),
                                  static_cast<T>(1.0), std::multiplies<T>());
-  std::transform(
-      m_hist.begin(), m_hist.end(), m_hist.begin(),
-      [tot_count, bin_volume](T v) { return v / (tot_count * bin_volume); });
+  std::transform(m_hist.begin(), m_hist.end(), m_hist.begin(),
+                 [this, bin_volume](T v) { return v / (bin_volume * static_cast<T>(m_tot_count));});
 }
 
-class CylindricalHistogram : public Histogram<double> {
+
+template <typename T>
+class CylindricalHistogram : public Histogram<T> {
 public:
-  using Histogram::Histogram;
+  using Histogram<T>::Histogram;
+  using Histogram<T>::get_n_bins;
+  using Histogram<T>::get_limits;
+  using Histogram<T>::get_bin_sizes;
+  using Histogram<T>::m_hist;
+  using Histogram<T>::m_tot_count;
```

Comment:
I don't think you need to `using` all these member functions. They should all come through inheritance already, the only exception being the constructor.

Reply:
Not true for template inheritance.